cmd/goal/clerk.go

```diff
@@ -473,6 +477,14 @@ var sendCmd = &cobra.Command{
 					Args:  programArgs,
 				},
 			}
+			var authAddr basics.Address
+			if signerAddress != "" {
+				authAddr, err = basics.UnmarshalChecksumAddress(signerAddress)
+				if err != nil {
+					reportErrorf("Signer invalid (%s): %v", signerAddress, err)
+				}
+				stx.AuthAddr = basics.Address(authAddr)
+			}
```

Comment:
Here again, I'm hoping we can remove some conditional logic, rather than add code to the `program != nil` case. But here I'm less confident.

Here it seems like the code that set `authAddr` could be unconditionally run before the whole `if/else if/else`.  Then the authAddr could be used here and below.

Reply:
Thanks for these points, I've updated the PR based on your feedback and ran a few tests manually to make sure it's still all working as expected.